ocpi-endpoints-msp-locations/src/main/scala/com/thenewmotion/ocpi/locations/MspLocationsRoute.scala

```diff
@@ -1,22 +1,24 @@
-package com.thenewmotion.ocpi.locations
+package com.thenewmotion.ocpi
+package locations
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.PathMatcher1
+import akka.http.scaladsl.server.Route
+import com.thenewmotion.mobilityid.CountryCode
+import com.thenewmotion.mobilityid.OperatorIdIso
+import msgs.v2_1.CommonTypes.SuccessResp
+import msgs.v2_1.CommonTypes.SuccessWithDataResp
+import msgs.v2_1.Locations._
+import msgs.v2_1.OcpiStatusCode._
 
-import akka.http.scaladsl.model.{HttpMethods, StatusCodes}
-import akka.http.scaladsl.server.{MethodRejection, PathMatcher1, Route}
-import com.thenewmotion.mobilityid.{CountryCode, OperatorIdIso}
-import com.thenewmotion.ocpi.msgs.v2_1.CommonTypes.{SuccessResp, SuccessWithDataResp}
-import com.thenewmotion.ocpi.msgs.v2_1.Locations._
-import com.thenewmotion.ocpi.{ApiUser, JsonApi}
-import org.joda.time.DateTime
 import scala.concurrent.Future
 import scalaz._
-import com.thenewmotion.ocpi.msgs.v2_1.OcpiStatusCode._
 
 class MspLocationsRoute(
-  service: MspLocationsService,
-  currentTime: => DateTime = DateTime.now
+  service: MspLocationsService
 ) extends JsonApi {
 
-  import com.thenewmotion.ocpi.msgs.v2_1.OcpiJsonProtocol._
+  import msgs.v2_1.OcpiJsonProtocol._
 
   private def leftToRejection[T](errOrX: Future[LocationsError \/ T])(f: T => Route): Route =
```

Comment:
have you considered using analog of spray marshallers support? Like following
implicit def eitherMarshaller[L, R](implicit lm: Marshaller[L], rm: Marshaller[R]) =
  Marshaller[\/[L, R]]{ (v, ctx) =>
    v.fold(lm(_, ctx), rm(_, ctx))
  }

implicit def eitherToResponseMarshaller[L, R](implicit lm: ToResponseMarshaller[L], rm: ToResponseMarshaller[R]) =
  ToResponseMarshaller[\/[L, R]]{ (v, ctx) =>
    v.fold(lm(_, ctx), rm(_, ctx))
  }